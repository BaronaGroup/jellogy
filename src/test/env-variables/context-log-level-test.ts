import {prepareTestKarhu} from '../test-utils'

describe('context-log-level-test', () => {
  const karhuTest = prepareTestKarhu({
    defaultLogLevel: 'ERROR',
    contextSpecificLogLevels: {
      ctx: 'ERROR'
    }
  })

  it('allows overriding default log level', karhuTest((karhu, output) => {
    karhu.context('ctx').warn('Should not show up')
    process.env.KARHU_LOG_LEVEL_ctx = 'WARN'
    karhu.context('ctx').warn('Should show up')
    process.env.KARHU_LOG_LEVEL_ctx = undefined
    expect(output.tracked).toMatchSnapshot()
  }))

  it('non-alphanumeric character sequences are handled as underscores in env variable name', karhuTest((karhu, output) => {
    const context = 'a+-/b    c'
    karhu.context(context).warn('Should not show up')
    process.env.KARHU_LOG_LEVEL_a_b_c = 'WARN'
    karhu.context(context).warn('Should show up')
    process.env.KARHU_LOG_LEVEL_a_b_c = undefined
    expect(output.tracked).toMatchSnapshot()
  }))
})
