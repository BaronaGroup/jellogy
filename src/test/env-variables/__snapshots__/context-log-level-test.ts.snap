// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context-log-level-test allows overriding default log level 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [
    Array [
      "0 WARN ctx[93m",
      "Should show up",
      "[39m",
    ],
  ],
  "ERROR" => Array [],
}
`;

exports[`context-log-level-test non-alphanumeric character sequences are handled as underscores in env variable name 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [
    Array [
      "0 WARN a+-/b    c[93m",
      "Should show up",
      "[39m",
    ],
  ],
  "ERROR" => Array [],
}
`;
